activerecord/test/cases/associations/cascaded_eager_loading_test.rb

```diff
@@ -161,6 +161,11 @@ def test_eager_association_loading_where_first_level_returns_nil
     end
   end
 
+  def test_eager_association_loading_with_missing_first_record
+    posts = Post.where(id: 3).preload(author: { comments: :post }).to_a
```

Comment:
It is not clear to me how this test is exercising this problem. Is the problem that The post with id 3 doesn't exist of if that the author that is missing, or the comment?

Reply:
@rafaelfranca The post with ID 3 has `author_id` nil, so `preloaders_for_hash` gets called with an empty `records` and `association` set to `{comments: :post}`